Framework/Algorithms/src/Bin2DPowderDiffraction.cpp

```diff
@@ -0,0 +1,385 @@
+#include "MantidAPI/FileProperty.h"
+#include "MantidAPI/InstrumentValidator.h"
+#include "MantidAPI/SpectrumInfo.h"
+#include "MantidAPI/SpectraAxisValidator.h"
+#include "MantidAPI/WorkspaceFactory.h"
+#include "MantidAPI/WorkspaceUnitValidator.h"
+#include "MantidKernel/CompositeValidator.h"
+#include "MantidAPI/WorkspaceProperty.h"
+#include "MantidKernel/UnitFactory.h"
+#include "MantidDataObjects/EventList.h"
+#include "MantidKernel/ArrayProperty.h"
+#include "MantidKernel/PropertyWithValue.h"
+#include "MantidKernel/RebinParamsValidator.h"
+#include "MantidKernel/VectorHelper.h"
+#include "MantidAPI/HistogramValidator.h"
+
+namespace Mantid {
+namespace Algorithms {
+
+using namespace Kernel;
+using namespace API;
+using namespace Geometry;
+using namespace DataObjects;
+using namespace Mantid::HistogramData;
+
+// Register the algorithm into the AlgorithmFactory
+DECLARE_ALGORITHM(Bin2DPowderDiffraction)
+
+//----------------------------------------------------------------------------------------------
+
+/// Algorithms name for identification. @see Algorithm::name
+const std::string Bin2DPowderDiffraction::name() const {
+  return "Bin2DPowderDiffraction";
+}
+
+/// Algorithm's version for identification. @see Algorithm::version
+int Bin2DPowderDiffraction::version() const { return 1; }
+
+/// Algorithm's category for identification. @see Algorithm::category
+const std::string Bin2DPowderDiffraction::category() const {
+  return "Diffraction\\Focussing";
+}
+
+/// Algorithm's summary for use in the GUI and help. @see Algorithm::summary
+const std::string Bin2DPowderDiffraction::summary() const {
+  return "Bins TOF powder diffraction event data in 2D space.";
+}
+
+//----------------------------------------------------------------------------------------------
+/** Initialize the algorithm's properties.
+ */
+void Bin2DPowderDiffraction::init() {
+  auto wsValidator = boost::make_shared<CompositeValidator>();
+  wsValidator->add<WorkspaceUnitValidator>("Wavelength");
+  wsValidator->add<SpectraAxisValidator>();
+  wsValidator->add<InstrumentValidator>();
+  wsValidator->add<HistogramValidator>();
+
+  declareProperty(
+      make_unique<WorkspaceProperty<EventWorkspace>>(
+          "InputWorkspace", "", Direction::Input, wsValidator),
+      "An input EventWorkspace must be a Histogram workspace, not Point data. "
+      "X-axis units must be wavelength.");
+
+  declareProperty(make_unique<WorkspaceProperty<API::Workspace>>(
+                      "OutputWorkspace", "", Direction::Output),
+                  "An output workspace.");
+
+  const std::string docString =
+      "A comma separated list of first bin boundary, width, last bin boundary. "
+      "Optionally "
+      "this can be followed by a comma and more widths and last boundary "
+      "pairs. "
+      "Negative width values indicate logarithmic binning.";
+  auto rebinValidator = boost::make_shared<RebinParamsValidator>(true);
+  declareProperty(
```

Comment:
This needs a better name. The user must look at the sphinx site in order to find out what it does. I suggest that this one be called `dSpaceBinning`.

Reply:
it seems that these names violate conventions, because RHEL7 build has failed with a message:
```
property (dSpaceBinning) violates conventions
property (dPerpendicularBinning) violates conventions
```